File: katas/6 kyu/backspaces-in-string/solution.py
# backspaces-in-string
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def clean_string(s):
  output = ''
  for letter in s:
    if letter == '#':
      output = output[:-1]
    else:
      output += letter
  return output
# ------------------------------ CLEVER SOLUTION ----------------------------- #
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(clean_string('abc#d##c'), "ac")
test.assert_equals(clean_string('abc####d##c#'), "" )
test.assert_equals(clean_string("#######"), "" )
test.assert_equals(clean_string(""), "" )

File: katas/6 kyu/backspaces-in-string/solution.js
// backspaces-in-string
/* ------------------------------- MY SOLUTION ------------------------------ */
function clean_string(s) {
  const result = []
  for (const c of s) {
    if (c === "#") {
      result.pop()
    } else {
      result.push(c)
    }
  }
  return result.join("")
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;

const {assert} = require('chai');

describe("cleanString", () => {
  it("should work correctly", () => {
    assert.strictEqual(cleanString('abc#d##c'), 'ac');
    assert.strictEqual(cleanString('abc####d##c#'), '');
  });
});
